libraries/AP_BattMonitor/AP_BattMonitor_SMBus.h

```diff
@@ -52,13 +57,36 @@ class AP_BattMonitor_SMBus : public AP_BattMonitor_Backend
     // return true if cycle count can be provided and fills in cycles argument
     bool get_cycle_count(uint16_t &cycles) const override;
 
+#if HAL_SMART_BATT_LOG_ENABLED
+    // returns true if serial number can be provided and fills in serial number
+    bool get_serial_number(char *serial_number, uint8_t buflen) const override;
+
+    // returns true if product name can be filled in from either manufacturer name and device name
+    bool get_product_name(char *product_name, uint8_t buflen) const override;
+
+    // returns true if design_capacity (capacity when newly manufactured) can be provided and fills it in
+    bool get_design_capacity(int32_t &design_capacity) const override;
+
+    // returns true if the full charge capacity (accounting for battery degradation) can be provided and fills it in
+    bool get_full_charge_capacity(int32_t &full_charge_capacity) const override;
+
+    // returns true if the design voltage (maximum charging voltage) can be provided and fill it in
+    bool get_design_voltage(int32_t &design_voltage) const override;
```

Comment:
These all need units. Design voltage should be a float in volts instead of integer millivolts, just to be consistent with the `voltage()` function. If you want to leave the design/charge capacity as mah (which is fine), then the names should end with `_mah`

Reply:
Added units in all the comments and made them volts and A-h. For the fields we are stashing in SMBus I kept the variables in mAh and mV and renamed them to be clear. 

For the design voltage in mV I would need to store it as float if it was in volts you decide?